Lib/textwrap.py

```diff
@@ -215,8 +215,14 @@ def _handle_long_word(self, reversed_chunks, cur_line, cur_len, width):
         # If we're allowed to break long words, then do so: put as much
         # of the next chunk onto the current line as will fit.
         if self.break_long_words:
-            cur_line.append(reversed_chunks[-1][:space_left])
-            reversed_chunks[-1] = reversed_chunks[-1][space_left:]
+            end = space_left
+            chunk = reversed_chunks[-1]
+            if self.break_on_hyphens and len(chunk) > space_left:
+                hyphen = chunk.rfind('-', 0, space_left)
+                if hyphen != -1:
```

Comment:
`"-1234567890"` would split as `"-"` and `"1234567890"`. Maybe check `hyphen > 0`?

Reply:
Good point!   Got me wondering about 1E-123456789 and 2**-123456789, but they are probably quite esoteric.

Reply:
If we have "--a_long_command_name" it will break right after the --.    To avoid that we could check that there is something other than '-' in chunk[0:hyphen].  Does that make sense?

Reply:
May be.